pkg/controller/logforwarding/logforwarding_controller.go

```diff
@@ -0,0 +1,153 @@
+package logforwarding
+
+import (
+	"context"
+	"fmt"
+
+	loggingv1 "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	loggingv1alpha1 "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_logforwarding")
+
+// Add creates a new LogForwarding Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileLogForwarding{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("logforwarding-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for creates only
+	pred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return true },
```

Comment:
did we decide that we're going to update every time? I thought we wanted to encourage users to stop updating the TP api and instead move to the GA one and this PR was to save them the headache of log loss upon upgrade.

to me, if we do this on updates then this isn't a "one-time conversion"

Reply:
I re-adjusted this one, because we stop reconciling to GA if a TP resource is mallformed. This allows users to fix their TP resource and let the controller convert it to a GA resource. The controller won't process any further update evengts if a GA resource was successfully created.

Reply:
if a TP resource is malformed does that mean that they wouldn't have logs flowing in as well? or is it possible that it can be malformed and work in TP?